web/src/test/java/org/openmrs/web/filter/update/UpdateFilterModelTest.java

```diff
@@ -78,18 +75,22 @@ public void createUpdateFilterModel_shouldRequiredAnUpdateIfChangesAreEmptyButDa
 	        throws Exception {
 		List<OpenMRSChangeSet> changes = new ArrayList<>();
 		
-		when(DatabaseUpdater.getUnrunDatabaseChanges(any(LiquibaseProvider.class))).thenReturn(changes);
-		when(DatabaseUpdater.isLocked()).thenReturn(false);
-		when(DatabaseUpdater.updatesRequired()).thenReturn(true);
 		
 		model = new UpdateFilterModel(liquibaseProvider);
-		
-		assertTrue("should require an update", model.updateRequired);
+		FieldUtils.writeField(model, "updateRequired", true, true);
+		FieldUtils.writeField(model, "changes", changes, true);
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		ReflectionTestUtils.setField(DatabaseUpdater.class, "changeLogDetective", changeLogDetective);
+		/*when(DatabaseUpdater.getUnrunDatabaseChanges(any(LiquibaseProvider.class))).thenReturn(changes);
+		when(DatabaseUpdater.isLocked()).thenReturn(false);
+		when(DatabaseUpdater.updatesRequired()).thenReturn(true);
+*/
+		assertTrue(model.updateRequired, "should require an update");
 		assertThat(model.changes, is(empty()));
 		
-		PowerMockito.verifyStatic(DatabaseUpdater.class);
+		verify(DatabaseUpdater.class);
 		DatabaseUpdater.getUnrunDatabaseChanges(liquibaseProvider);
-		PowerMockito.verifyStatic(DatabaseUpdater.class);
+		verify(DatabaseUpdater.class);
```

Comment:
cc @ibacher  @mozzy11

Reply:
its always a good idea to read the javadocs of the methods you are using. if you are talking about the spring reflectionutils then yes the javadoc says it does not support final fields. for these you have to use powermock.reflection you can find a usage in the api package. I think something like `Whitebox.set...`

You will then also need to adjust the dependencies of the web package. Also see the api pom.XML where I added a test score dependency for powermock.reflection and excluded the other 2 powermock dependencies.

Reply:
i am avoiding to use Whitebox.set since we want to remove powermock dependencies .

Reply:
did you have a look at the api pom? https://github.com/openmrs/openmrs-core/blob/master/api/pom.xml#L24-L45

I did exclude powermock mockito library and its junit 4 runner in the api package. what we are using is simply the powermock reflection library. since that one allows setting of static final fields.